Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+
+                _context.Add(branch);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            //var branch = await _context.Branches.FindAsync(id);
+
+            if (branch == null)
+            {
+                return NotFound();
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // POST: Branches/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, BranchFormViewModel branchVM)
+        {
+            if (id != branchVM.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    var branch = _mapper.Map<Branch>(branchVM);
+                    _context.Update(branch);
+                    await _context.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!BranchExists(branchVM.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Delete/5
+        public async Task<IActionResult> Delete(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (branch == null)
+            {
+                return NotFound();
+            }
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // POST: Branches/Delete/5
+        [HttpPost, ActionName("Delete")]
```

Comment:
Why not simply [HttpDelete]

Reply:
then you don't have from in your view but button with action link

Reply:
@FabloAlencar, this code you did is working perfectly fine, but there are some other reasons why we wish to use proper verbs like [HttpDelete] attribute point to. There is HTML semantic important. POST form indicate resource creation, but you actualy remove resource. Microsoft of course did bad example like here (https://docs.microsoft.com/pl-pl/aspnet/mvc/overview/getting-started/introduction/examining-the-details-and-delete-methods).
Please read this:
https://maxtoroq.github.io/2012/08/implementing-delete-button-in-aspnet-mvc.html
and make use of it to implement semantically correct HTML code

Reply:
Lets leave it for now. I would search for it more.